tests/integration/modifiers/real-did-intersect-test.js

```diff
@@ -0,0 +1,48 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+module(
+  'Integration | Modifier | did-intersect without mocks',
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    hooks.beforeEach(function () {
+      this.enterStub = sinon.stub();
+      this.exitStub = sinon.stub();
+      this.maxEnter = 1;
+      this.maxExit = 1;
+
+      this.observerManagerService = this.owner.lookup(
+        'service:ember-scroll-modifiers@observer-manager',
+      );
+    });
+
+    test('it removes elements from the registry on modifier cleanup', async function (assert) {
+      let removedElementsCount = 0;
+      let originalRemoveElement = this.observerManagerService.removeElement;
+      const observerManagerService = this.observerManagerService;
+
+      observerManagerService.removeElement = function removeElementExt(
```

Comment:
I think this would be better accomplished with `sinon.spy` - it will still call the underlying function and then you can check the call count on the spy in your assertion.